src/pages/iv-items/forms/KitForm.js

```diff
@@ -133,40 +141,24 @@ const KitForm = ({ store, labels, maxAccess }) => {
     }
   ]
 
-  useEffect(() => {
-    setNumRows(formik?.values?.kit?.length)
-  }, [formik.values.kit])
-
   function getData() {
     getRequest({
       extension: InventoryRepository.Kit.qry,
       parameters: `_kitId=${recordId}`
+    }).then(res => {
+      const modifiedList = res.list?.map((kitItems, index) => ({
+        ...kitItems,
+        id: index + 1
+      }))
+      modifiedList.length > 0 && formik.setValues({ kit: modifiedList })
     })
-      .then(res => {
-        const modifiedList = res.list?.map((kitItems, index) => ({
-          ...kitItems,
-          id: index + 1
-        }))
-        formik.setValues({ kit: modifiedList })
-      })
-      .catch(error => {})
   }
   useEffect(() => {
```

Comment:
use  useEffect(() => {
    ;(async function () {   })()
  }, [])  and remove dependency recordId